iceoryx_hoofs/source/concurrent/loffli.cpp

```diff
@@ -42,7 +42,7 @@ void LoFFLi::init(cxx::not_null<Index_t*> freeIndicesMemory, const uint32_t capa
         for (uint32_t i = 0; i < m_size + 1; i++)
         {
             // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic) upper limit of index is set by m_size
-            m_nextFreeIndex[i] = i + 1;
+            m_nextFreeIndex.get()[i] = i + 1;
```

Comment:
I will review later, but just looking at this syntax it already feels wrong to remove the operator. It is not really clear from reading either, just more verbose and confusing.

I clearly remember why I opted for the implicit conversion to pointer. If I read what is originally there I just think: "sure, it is some kind of pointer and we use pointer arithmetic and dereference it." For the understanding at this level it does not matter what kind of pointer it has, only that it has pointer syntax (and the corresponding access semantics).

But now I pause a little and need to think about that I probably get some pointer back and then we perform the pointer arithmetic. It would even look worse with explicit conversion.

For me it is the posterchild case where implicit conversion is the way to go.

Reply:
Sure, making it explicit is worse and I do not advocate this. I advocate keeping it implicit as it shall behave like a pointer from the point of view of the client,  which is exactly what is accomplished by implicit conversion.